weldx/asdf/schemas/weldx.bam.de/weldx/datamodels/single_pass_weld-1.0.0.schema.yaml

```diff
@@ -17,52 +26,406 @@ properties:
-      seam_length:
+          The base metal composition of the workpiece.
+        type: object
+        properties:
+          common_name:
+            description: |
+              The common description of the base metal composition or classification as listed in the standard.
+            type: string
+          standard:
+            description: |
+              The standard listing and describing the base metal compositions.
+            type: string
+        required: [common_name, standard]
+      geometry:
         description: |
-          Length of the linear weld seam.
-        tag: "tag:stsci.edu:asdf/unit/quantity-1.1.0"
-        wx_unit: "m"
-    required: [groove_shape, seam_length]
+          Description of the workpiece geometry consisting of the groove shape and the total seam length.
+        type: object
+        properties:
+          groove_shape:
+            description: |
+              Constant groove shape of the weld seam.
+            oneOf:
+              - tag: "tag:weldx.bam.de:weldx/groove/iso_9692_1_2013_12/DHUGroove-1.0.0"
+              - tag: "tag:weldx.bam.de:weldx/groove/iso_9692_1_2013_12/DHVGroove-1.0.0"
+              - tag: "tag:weldx.bam.de:weldx/groove/iso_9692_1_2013_12/DUGroove-1.0.0"
+              - tag: "tag:weldx.bam.de:weldx/groove/iso_9692_1_2013_12/DVGroove-1.0.0"
+              - tag: "tag:weldx.bam.de:weldx/groove/iso_9692_1_2013_12/FFGroove-1.0.0"
+              - tag: "tag:weldx.bam.de:weldx/groove/iso_9692_1_2013_12/HUGroove-1.0.0"
+              - tag: "tag:weldx.bam.de:weldx/groove/iso_9692_1_2013_12/HVGroove-1.0.0"
+              - tag: "tag:weldx.bam.de:weldx/groove/iso_9692_1_2013_12/IGroove-1.0.0"
+              - tag: "tag:weldx.bam.de:weldx/groove/iso_9692_1_2013_12/UGroove-1.0.0"
+              - tag: "tag:weldx.bam.de:weldx/groove/iso_9692_1_2013_12/UVGroove-1.0.0"
+              - tag: "tag:weldx.bam.de:weldx/groove/iso_9692_1_2013_12/VGroove-1.0.0"
+              - tag: "tag:weldx.bam.de:weldx/groove/iso_9692_1_2013_12/VVGroove-1.0.0"
+          seam_length:
+            description: |
+              Length of the linear weld seam.
+            tag: "tag:stsci.edu:asdf/unit/quantity-1.1.0"
+            wx_unit: "m"
+        required: [groove_shape, seam_length]
+    required: [base_metal, geometry]
   reference_timestamp:
+    description: |
+      An optional timestamp indicating the start of the welding process.
     tag: "tag:weldx.bam.de:weldx/time/timestamp-1.0.0"
-  meta:
+  wx_metadata:
     description: |
       General metadata container.
     type: object
-required: [equipment,geometry,measurements,coordinate_systems,welding_current,welding_voltage]
+  wx_user:
+    description: |
+      Metadata container for additional user documentation of the experiment.
+    type: object
+required: [equipment,workpiece,measurements,welding_current,welding_voltage,TCP]
+additionalProperties: false
+
+
+examples:
```

Comment:
encoding an example is very nice. However I'm unsure, if a schema is the right spot for this. Wouldn't it be clearer (for the users of the schemes at least), if examples are placed in some example folder?

Reply:
examples that are included in the schemas like this get integrated into the documentation (and tested even, although disabled in this case) [like so](https://weldx--256.org.readthedocs.build/en/256/generated/weldx.bam.de/weldx/datamodels/single_pass_weld-1.0.0.schema.html#Examples)

Reply:
yes that's useful to include it in the (developer) documentation. I just have the feeling that users would be more interested in the example without the schema, although they would likely never write a schema themself but stick to the Python api.

Reply:
Yes we shall cover those in detail with some extensive reference files